tag.go

```diff
@@ -105,6 +105,27 @@ func (t *Tag) Decode(o core.Object) (err error) {
 	return nil
 }
 
+// Encode transforms a Tag into a core.Object.
+func (t *Tag) Encode(o core.Object) error {
+	o.SetType(core.TagObject)
+	w, err := o.Writer()
+	if err != nil {
+		return err
+	}
+	w.Write([]byte("object "))
+	w.Write([]byte(t.Target.String()))
+	w.Write([]byte("\ntype "))
+	w.Write(t.TargetType.Bytes())
+	w.Write([]byte("\ntag "))
+	w.Write([]byte(t.Name))
+	w.Write([]byte("\ntagger "))
+	t.Tagger.Encode(w)
+	w.Write([]byte("\n\n"))
+	w.Write([]byte(t.Message))
+	o.Hash()
```

Comment:
It's the only way to ensure that the object has a computed hash.
If we want to change it, we could introduce a Finalize method, or something like that, that can be called once the object is written to complete any further fields (e.g. hash, size).

It seems to me that changing that is way beyond the scope of this PR though.